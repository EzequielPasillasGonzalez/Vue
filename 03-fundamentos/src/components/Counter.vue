<template>
<div>


        <!-- <h2> {{ titulo || 'Counter' }} </h2> --> <!-- ? Aqui se le asigna el valor de 'Counter' si viene vacio el tituloS -->

    
    <h2  > {{ customTitle }}</h2>    

    <p>{{ counter }} <sup>2</sup> = {{ squareCounter }}  </p>

    <div class="button">        
        <button @click="decrease">-1</button>
        <button @click="increase">+1</button>
    </div>


</div>
</template>

<script>
export default {

    // La diferencia de los atributos y las porpertis es que las properti ya estan previamente definidas 
    //props:['titulo', 'start'], // Asi se reciben las propertis
    props:{ // Las propiedades que acepta el componente
        titulo: String,
        start: {

            type: Number,
            
            //require: true // Hacer que la properti sea obligatoria
            
            default: 100, // POner un valor por defecto

            validator(value){ // Hacer validaciones del valor que se recibe
                return value >= 0
            }
        }
    },  

    data(){
        return {
            counter: this.start
        } 
    },

    methods: {
        
        getSquareValue(){            
            return this.counter // Hace referencia al valor que tiene en la data
        },

        increase(){            
            this.counter ++            
        },

        decrease(){            
            this.counter --            
        }
    },
    computed:{
        squareCounter(){ // Es mas practico por si se necesita realizar el mismo proceso muchas veces            
            return this.counter * this.counter
        },

        customTitle(){
            if(this.titulo === undefined){
                return 'Counter'
            }
            
            return this.titulo
            
        }
    }

}
</script>

<style>

    button {
        background-color:  #64BB87;
        border-radius:  5px;
        border: 1px solid white;
        color: white;
        margin: 0 5px;
        padding: 5 px 15px;
        transition:  0.3s ease-in-out;
    }

    button:hover{
        background-color:  #5aa67b;
        transition: 0.3s ease-in-out;
    }

    #titulo{
        font-weight: bold;
    }

</style>