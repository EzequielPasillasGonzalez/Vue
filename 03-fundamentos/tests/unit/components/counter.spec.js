import { mount, shallowMount} from '@vue/test-utils' //? el Mount monta todos los componentes y subcomponentes y el shallowmont monta un componente 
import Counter  from "@/components/Counter"; //? Se trae al componente para ser evaluado

describe('Counter Component', () => {  //? Es el nombre que se le da a la prueba

    //? Hacer una revision por si se cambio algo en el codigo
    // test('Debe de hacer match con el snapshot', () =>{

    //     const wrapper = shallowMount(Counter) //? Es el metodo con el que se montan los componentes

    //     expect(wrapper.html()).toMatchSnapshot() //? se evalua oara ver si no se ha cambiado el html del componente

    // })    

    test('H2', () => { second })



})