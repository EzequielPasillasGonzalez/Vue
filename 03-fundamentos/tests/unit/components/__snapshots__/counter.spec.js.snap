// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter Component Debe de hacer match con el snapshot 1`] = `
<div>
  <!-- <h2> {{ titulo || 'Counter' }} </h2> -->
  <!-- ? Aqui se le asigna el valor de 'Counter' si viene vacio el tituloS -->
  <h2>Counter</h2>
  <p>100 <sup>2</sup> = 10000</p>
  <div class="button"><button>-1</button><button>+1</button></div>
</div>
`;
